client/app/routes/collection.js

```diff
@@ -2,8 +2,8 @@ import Ember from 'ember';
 
 export default Ember.Route.extend({
   model (params) {
-    return this.store.findRecord('collection', params.collection_id).then(function(data){
-      return data;
-    });
-  }
+        return this.store.findRecord('collection', params.collection_id).then(data => data).catch(error => {
+            this.transitionTo('not-found');
```

Comment:
When I go to a collection that does not exist, the page doesn't load and I see the error message `Uncaught TypeError: Cannot read property 'name' of undefined` in the console.